salt/transport/tcp.py

```diff
@@ -647,9 +591,10 @@ def publish_payload(self, package):
 
 
 class TCPPubServerChannel(salt.transport.server.PubServerChannel):
-    def __init__(self, opts):
+    def __init__(self, opts, io_loop=None):
         self.opts = opts
         self.serial = salt.payload.Serial(self.opts)  # TODO: in init?
+        self.io_loop = io_loop or tornado.ioloop.IOLoop.current()
 
     def _publish_daemon(self):
```

Comment:
Since this is now on the IOLoop we don't even need a separate process for this. If we want to maintain one we should keep our own ioloop instead of start() ing one from the base process to avoid weirdness